### PR TITLE
Tensorize bases [tbasis-dev]

### PR DESCRIPTION
**Pull request by jdahm**
Inherit from `NodalTensorBasisElement` or `PositiveTensorBasisElement` (which themselves inherit from `TensorBasisElement`) when the basis can be represented by a tensor product of 1D bases. This commit is a standalone change made in the occa-dev branch and should be useful for other parts of the code as well. We may still need to work on tensorizing more of the bases before merging.

### CODE REVIEW

File: fem/fe.hpp [hunk 1]
```diff
@@ -1364,11 +1445,18 @@ class RotTriLinearHexFiniteElement : public NodalFiniteElement
 class Poly_1D
 {
 public:
+   enum
+   {
+     ChangeOfBasis = 0, // Use change of basis, O(p^2) Evals
+     Barycentric   = 1, // Use barycentric Lagrangian interpolation, O(p) Evals
+     Positive      = 2, // Use bernstein polynomials
```
Comment (YohannDudouit) [inline fem/fe.hpp]: Why Bernstein polynomials don't fall under the Change of Basis type?

File: fem/fe.cpp [hunk 2]
```diff
@@ -6566,61 +6591,285 @@ const double *Poly_1D::GetPoints(const int p, const int type)
...
+            dof_map[p + (p + 0*p1)*p1] = 2;
+            dof_map[0 + (p + 0*p1)*p1] = 3;
+            dof_map[0 + (0 + p*p1)*p1] = 4;
+            dof_map[p + (0 + p*p1)*p1] = 5;
+            dof_map[p + (p + p*p1)*p1] = 6;
+            dof_map[0 + (p + p*p1)*p1] = 7;
+
+            // edges (see Hexahedron::edges in mesh/hexahedron.cpp)
+            int o = 8;
+            for (int i = 1; i < p; i++)
```
Comment (YohannDudouit) [inline fem/fe.cpp]: Shouldn't we replace this with a macro or an inline function? If there is ever a bug in this, we're unlikely to find it. Macros are not safe in C++ though.

**Comment by tzanio** [comment]
@YohannDudouit any objections to merge this?

**Comment by tzanio** [comment]
Mention in `CHANGELOG`?

File: CHANGELOG [hunk 3]
```diff
@@ -23,6 +26,7 @@ Miscellaneous
 - Upgraded MFEM to support Axom 0.2.8. Prior versions are no longer supported.
 
 - Add support for building a shared version of the MFEM library with GNU make.
+>>>>>>> master
```
Comment (v-dobrev) [inline CHANGELOG]: Oops.

Comment (tzanio) [reply CHANGELOG]: Fixed

**Pull request merged**